src/tike/opt.py

```diff
@@ -234,11 +234,14 @@ def direction_dy(xp, grad0, grad1, dir_):
         The previous search direction.
 
     """
-    return (
-        - grad1
-        + dir_ * xp.linalg.norm(grad1.ravel())**2
-        / (xp.sum(dir_.conj() * (grad1 - grad0)) + 1e-32)
-    )  # yapf: disable
+    if dir_ is None:
+        return [-g for g in grad1]
+
+    return [
+        - grad1[0]
+        + dir_[0] * xp.linalg.norm(grad1[0].ravel())**2
+        / (xp.sum(dir_[0].conj() * (grad1[0] - grad0[0])) + 1e-32)
+    ]  # yapf: disable
```

Comment:
What's a better way to do this that doesn't involve assuming grad1 is both a list of any length and a list of length 1?